### PR TITLE
Refactor kube2sky.go to make it usable from other packages.

### PR DESCRIPTION
**Pull request by dlorenc**
## Pull Request Guidelines
1. Please read our [contributor guidelines](https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md).
2. See our [developer guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md).
3. Follow the instructions for [labeling and writing a release note for this PR](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes) in the block below.

``` release-note
* Use the release-note-* labels to set the release note state 
* Clear this block to use the PR title as the release note 
-OR-
* Enter your extended release note here
```

[Analytics]()

Most logic moved to lib/kube2sky_lib.go. The flag parsing logic
remains in kube2sky.go. Some structs and types were updated to
make this feasible.

cc @luxas @vishh @ethernetdan
This is for the minikube project.

### CODE REVIEW

**Comment by luxas** [comment]
@dlorenc You should update `RELEASES.md` and bump the `TAG`

**Comment by vishh** [comment]
@luxas Why? Is this PR introducing any functional changes?

**Comment by luxas** [comment]
@vishh No, but the image isn't exactly as the one before and should therefore not be force pushed.
If a force push of the image would be made, it might (will) break existing clusters

**Comment by vishh** [comment]
I think the policy is to **not** update existing tags ever.

**Comment by luxas** [comment]
A force push should absolutely not be made!
I just answered your question why he should update the tag

TL;DR; @dlorenc bump the tag and explain what changes :)

**Comment by dlorenc** [comment]
> TL;DR; @dlorenc bump the tag and explain what changes :)

Done, thanks for explaining the tag/build system here.

File: cluster/addons/dns/kube2sky/Changelog [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## Version 1.16 (May 3 2016 Dan Lorenc<lorenc.d@gmail.com>)
+ - No behavior changes since 1.15
+ - Refactored kube2sy.go to allow use from other packages.
```
Comment (luxas) [inline cluster/addons/dns/kube2sky/Changelog]: nit: kube2sky

**Comment by luxas** [comment]
Also see @ArtfulCoder's work here: #23930

File: cluster/addons/dns/kube2sky/Changelog (hunk 1 above)
Comment (dlorenc) [inline cluster/addons/dns/kube2sky/Changelog]: Fixed. Nice catch!

**Comment by thockin** [comment]
@ArtfulCoder has a near-full rebuild of DNS in flight - can you explain what you want to do with this?

**Comment by dlorenc** [comment]
> @ArtfulCoder has a near-full rebuild of DNS in flight - can you explain what you want to do with this?

I'm happy to wait for that PR to land if that's easier. I got confused reading the description and thought it wasn't going to be merged ever.

This was to help out redspread/localkube which currently has a fork of the existing skydns code here: https://github.com/redspread/localkube/blob/master/k2s/kube2sky.go

This PR would help remove that fork by moving the flag parsing/func main logic to a separate package, making the main logic directly importable.
